Code/Framework/AzToolsFramework/AzToolsFramework/ComponentMode/ComponentModeSwitcher.h

```diff
@@ -103,7 +104,7 @@ namespace AzToolsFramework
                 const EntityIdList& newlySelectedEntities, const EntityIdList& newlyDeselectedEntities) override;
 
             // Member variables
-            ComponentData* m_activeSwitcherComponent = nullptr; //!< The component that is currently in component mode
+            AZ::Component* m_activeSwitcherComponent = nullptr; //!< The component that is currently in component mode
```

Comment:
comment: Just be really careful with storing pointers to Components as this can break horribly with undo/redo in the Editor - might be better to store the EntityComponentId pair instead and resolve the Component when you need it.